CHANGELOG.md

```diff
@@ -6,7 +6,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
-- Nothing yet!
+- Fix theoretical thread-safety issue for concurrent calls to an uncached
+  zero argument method
```

Comment:
```suggestion
- Fix thread-safety issue in concurrent calls to zero-arg method in unmemoized
  state which resulted in a `nil` value being accidentally returned in one thread
```

Reply:
Apologies for the nit-pick, but I would say this issue was demonstrably _not purely theoretical_ 😄